packages/headless/src/modules/store/services/index.ts

```diff
@@ -0,0 +1,80 @@
+import { HttpStatus, Injectable } from '@nestjs/common';
+import { Helper } from 'src/helper/helper.interface';
+import { StoreRepository } from '../repositories';
+import {
+  CreateStoreErrorMessages,
+  CreateStoreRequestBody,
+  CreateStoreResponse,
+} from 'models';
+import { randomUUID } from 'crypto';
+import * as bcrypt from 'bcrypt';
+import { authConfig } from 'config/auth';
+
+@Injectable()
+export class StoreService {
+  constructor(private storeRepo: StoreRepository, private helper: Helper) {}
+
+  async createStore(
+    data: CreateStoreRequestBody,
+  ): Promise<CreateStoreResponse> {
+    const doesStoreShopOrLegalNameMatch =
+      (await this.storeRepo.getStore({
+        'info.shopName': data.info.shopName,
+      })) ||
+      (await this.storeRepo.getStore({
+        'info.legalName': data.info.legalName,
+      }));
+    if (doesStoreShopOrLegalNameMatch)
+      return this.helper.serviceResponse.errorResponse(
+        CreateStoreErrorMessages.STORE_SHOP_OR_LEGAL_NAME_EXISTS,
+        null,
+        HttpStatus.BAD_REQUEST,
+      );
```

Comment:
generate a url from shop name and save it in lower case letter. also consider it as unique.
make legalname optional